### ISSUE TITLE
Component Testing with Angular & Vite

### ISSUE THREAD

**Issue by leimerp** [cypress-io/cypress#30320]
### What would you like?

With the release of Angular 17, Vite has become an officially supported bundler, and many Angular applications are transitioning from Webpack to Vite. To maintain consistency in the development stack, it would be beneficial if Cypress supported Vite for component testing in Angular applications. This would allow developers to use the same bundler for their application, e2e tests, and component tests, streamlining the workflow and improving efficiency.

### Why is this needed?

Providing Vite support for component testing in Angular applications would align with the growing adoption of Vite and ensure a unified testing and development experience. It would help developers avoid potential bundling discrepancies and reduce the complexity of managing different bundlers for various testing scenarios.

**Comment by dowoncha** [comment]
Hello, are there any plans to update this? Does it require a specific config to be passed?

**Comment by SalvatoreAmoroso** [comment]
I agree that would be great otherwise you need to install both vite and webpack to use cypress component testing with modern angular

**Comment by etkr** [comment]
I would also like to add that Esbuild + Vite is now the default bundler-dev-server combo when generating a new Angular application as of atleast version 19. 

Additionally the Angular teams recommends migrating existing application with an automated migration script.

https://angular.dev/tools/cli/build-system-migration#automated-migration-recommended

It really seems like the Angular team is doing everything they can to get people to move away from Webpack.

Anecdotally the product I work is currently migrating to Esbuild and Vite and Cypress component tests are the only place where Webpack is still being used.

**Comment by Andreasgdp** [comment]
Given the continued and increasing adoption of Vite in the Angular ecosystem (now the default bundler), this feature remains highly relevant for many of us.

Could a maintainer please provide an update on this? Specifically, it would be great to understand:

- What are the current plans or thoughts around supporting Vite for Angular component testing?

- Are there any known blockers or specific challenges that need to be addressed?

- Is this an area where the community could contribute? If so, could you point us in the right direction or suggest where to start looking into the codebase?

Many of us are keen to help move this forward and would appreciate any guidance.

Thanks!

**Comment by waterplea** [comment]
We ran into an issue where old webpack builder compiles styles differently, effectively making our test environment qualitatively differ from actual usage which is a problem. Hopefully, the Cypress team would address this sooner rather than later 🤞

**Comment by ud4y-sc** [comment]
Any update on when vite/esbuild support will be added to Angular component tests?

**Comment by HHristovIV** [comment]
Hello all!

I am also interested in what the plan for adopting vite for angular component tests is? We are currently stuck with using a third-party adapter library to be able to get the tests working with the vite bundler in angular 21.

Thanks in advance,
Hristo

**Comment by jwheap** [comment]
Any updates on this?  We have an Nx monorepo with lots of different apps.  We found that the vast majority of time running our tests in our CI/CD pipelines is used for Webpack to create a new devServer when a new Cypress component test file is launched.  We created a custom script to merge all component test files into a single file (per lib), which definitely helped the overall time.  Vite is so, so, so much faster than Webpack, and that move alone would most likely eliminate the need for custom scripts and such to help reduce the time Webpack needs to build the devServer per file.  I know there are third-party packages (ones that can help with E2E) but we have many more component tests than E2E tests, so seeing Vite incorporated officially into Cypress component tests for Angular would be awesome.

Thank you to all the developers and maintainers here.  Your work is appreciated!
